src/lib.rs

```diff
@@ -105,6 +105,9 @@ pub mod utils;
 mod types;
 pub use crate::types::*;
 
+#[cfg(feature = "async")]
+pub mod r#async;
```

Comment:
Maybe it is worth renaming `async.rs` to `async_impl.rs` or something? Just to avoid this "keyword identifier" usage as it seems to add some complexity into `use` expressions.

Reply:
I renamed this